### PR TITLE
Add token balances endpoint (`/analytics/balances`)

### PR DESCRIPTION
**Pull request by jefag**
## PR-Codex overview
The focus of this PR is to add functionality to fetch treasury balances based on different time frequencies.

### Detailed summary
- Added `treasury` array to `TenantToken` type in `types.d.ts`
- Defined `TREASURY` constant in `uniswap.ts`
- Implemented `getTreasuryBalanceTS` function in `route.ts` to fetch treasury balances
- Added API endpoint to retrieve treasury balances based on frequency

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jefag** [comment]
# Routes

* http://localhost:3000/api/treasury_balance/24h
* http://localhost:3000/api/treasury_balance/7d
* http://localhost:3000/api/treasury_balance/1mo
* http://localhost:3000/api/treasury_balance/3mo
* http://localhost:3000/api/treasury_balance/1y

# Example Payload for the `3mo` endpoint...

```
{
    "result": [
        {
            "day": "2024-03-01T00:00:00.000Z",
            "date": "2024-03-01",
            "ts": "1709251200",
            "balance_usd": "4115630565"
        },
        {
            "day": "2024-06-01T00:00:00.000Z",
            "date": "2024-06-01",
            "ts": "1717200000",
            "balance_usd": "3825185518"
        }
    ]
}
```

**Comment by jefag** [comment]
### Questions for Reviewers

1. Which date type do you want? I'm supplying 3 formats.  Can you pick one that works best?  I'd like to remove the ones not in use.
2. Would naming the field `value` instead of `balance_usd` be helpful in abstracting what I send you?
3. Do you think we should check auth on this endpoint?  It's just giving away public data.  
4. Notice I added a `.treasury` property to the `Tenant` configs.  Any concerns there?

**Comment by jefag** [comment]
6. Is this a smart spot to put this endpoint?  
7. Should I nest is deeper down?
8. Where should I document an endpoint like this?

**Review by andreitr — APPROVED** [review]
LGTM

**Comment by stepandel** [comment]
hey @jefag what is the plan to consume this data? Is it only expected to be available in the app or you're planning to plug into it externally?
If the former, we have a pattern to use serve actions for internal consumptions so that we don't expose unnecessary endpoints publicly. Here's the example: https://github.com/voteagora/agora-next/blob/main/src/app/api/common/proposals/getProposals.ts

If the latter -- let's protect the API with the API key

File: src/app/api/treasury_balance/[frequency]/route.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                AND day >= (CURRENT_DATE - INTERVAL '${lookback} day')
+                AND ${skipCrit}
+                AND address IN ${crit}
+              GROUP BY 1 
+              ORDER BY day`;
+
+  const result = await prisma.$queryRawUnsafe<TokenBalance[]>(QRY);
+
+  return { result };
+}
```
Comment (stepandel) [inline src/app/api/treasury_balance/[frequency]/route.ts:73]: would be great to refactor these into `/app/api/balances`

Comment (stepandel) [inline src/app/api/treasury_balance/[frequency]/route.ts:73]: would be great to refactor these into `/app/api/balances`

Comment (jefag) [reply src/app/api/treasury_balance/[frequency]/route.ts]: Done. [57292e3](https://github.com/voteagora/agora-next/pull/355/commits/57292e32687906139a0e7458d828d2c0d1c6b8b1)

Comment (jefag) [reply src/app/api/treasury_balance/[frequency]/route.ts]: Done. [57292e3](https://github.com/voteagora/agora-next/pull/355/commits/57292e32687906139a0e7458d828d2c0d1c6b8b1)

**Comment by jefag** [comment]
> hey @jefag what is the plan to consume this data? Is it only expected to be available in the app or you're planning to plug into it externally? If the former, we have a pattern to use serve actions for internal consumptions so that we don't expose unnecessary endpoints publicly. Here's the example: https://github.com/voteagora/agora-next/blob/main/src/app/api/common/proposals/getProposals.ts
> 
> If the latter -- let's protect the API with the API key

I protected with auth.    Need to test still, but I copy-pasta is done.  This is what you expected, right?

[5a1248d](https://github.com/voteagora/agora-next/pull/355/commits/5a1248de7e87b828a732c795e112c0652e3c5239)

**Pull request merged**